### PR TITLE
Add a flag into RendererOptions to prevent render_on_scroll

### LINKED ISSUE #1497: Need a way to disable RenderBackend::render_on_scroll
Gecko with webrender does not need "render on scroll", it just spend cpu time. It is nice if we could disable it.

Gecko expects to generate new frames only with ApiMsg::GenerateFrame, since gecko generates new frames with v-sync timings. It is done by [WebRenderBridgeParent::CompositeToTarget()](https://dxr.mozilla.org/mozilla-central/source/gfx/layers/wr/WebRenderBridgeParent.cpp#1082). Frame scrolling is also controled by CompositeToTarget(). Then there is no necessity to do render on scroll by webrender. It just consumes cpu time.

gecko already does not handle new_scroll_frame_ready event by [Bug 1377688](https://bugzilla.mozilla.org/show_bug.cgi?id=1377688).

### PR DESCRIPTION
**Pull request by sotaroikeda**
Fixes #1497

cc  @kvark

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/1521)

### CODE REVIEW

File: webrender/src/render_backend.rs [hunk 1]
```diff
@@ -433,7 +437,9 @@ impl RenderBackend {
                                 });
                             }
 
-                            self.render_on_scroll = true;
+                            if self.enable_render_on_scroll {
+                                self.render_on_scroll = true;
```
Comment (kvark) [inline webrender/src/render_backend.rs]: could just be `self.render_on_scroll = self.enable_render_on_scroll`

File: webrender/src/render_backend.rs [hunk 2]
```diff
@@ -121,6 +124,7 @@ impl RenderBackend {
             vr_compositor_handler,
             window_size: initial_window_size,
             inner_rect: DeviceUintRect::new(DeviceUintPoint::zero(), initial_window_size),
+            enable_render_on_scroll: enable_render_on_scroll,
```
Comment (kvark) [inline webrender/src/render_backend.rs]: nit: could just be `enable_render_on_scroll,`

Comment (sotaroikeda) [inline webrender/src/render_backend.rs]: Thanks, I'll update it in a next patch.

File: webrender/src/render_backend.rs (hunk 1 above)
Comment (sotaroikeda) [inline webrender/src/render_backend.rs]: Yea, I'll update it in a next patch.

**Comment by kvark** [comment]
Thank you!
@bors-servo r+

**Pull request merged**